Regression in 5.109.0: resolve.alias mapping to an absolute path that points at a package directory no longer resolves

### Have you used AI?

Yes

### Bug Description

Using resolve.alias to make an alias to an absolute path of a package inside node_modules, webpack 5.109.0+ fails to resolve the package with the following error:

```
Module not found: Error: Can't resolve 'vendor/tippy.js'
```

This worked in 5.108.4

### Link to Minimal Reproduction and step to  reproduce

```
// webpack.config.js
const path = require('path');
module.exports = {
  mode: 'development',
  entry: './index.js',
  resolve: {
    alias: { vendor: path.resolve(__dirname, 'node_modules') },
  },
};

// index.js
import('vendor/tippy.js');
```

Build with webpack 5.109.0/5.109.1 → can't resolve 'vendor/tippy.js' error
Build with webpack 5.108.4 → succeeds

### Expected Behavior

`vendor/tippy.js` should resolve using the configured alias to `<abs>/node_modules/tippy.js` as it did in 5.108.4


### Actual Behavior

Can't resolve error

### Environment

```shell
- Webpack: first-bad at 5.109.0 (also bad at 5.109.1); last-good 5.108.4
- enhanced-resolve: 5.24.3 (identical on both)
- Node.js: v24.18.0 and v26.5.0
- OS: macOS (darwin)
- verified by installing 5.108.4 / 5.109.0 / 5.109.1 side-by-side and running the same config + entry against each
```

### Is this a regression?

Yes (please specify version below)

### Last Working Version

5.108.4

### Additional Context

Investigated with the help of claude, reviewed humanly before posting

--- a/.changeset/020-extension-alias-package-directory.md
+++ b/.changeset/020-extension-alias-package-directory.md
@@ -1,0 +1,5 @@
+---
+"webpack": patch
+---
+
+Resolve aliases pointing at a package directory whose name ends with `.js` again.

--- a/package.json
+++ b/package.json
@@ -105,7 +105,7 @@
     "acorn": "^8.16.0",
     "browserslist": "^4.28.1",
     "chrome-trace-event": "^1.0.2",
-    "enhanced-resolve": "^5.24.2",
+    "enhanced-resolve": "^5.24.4",
     "es-module-lexer": "^2.1.0",
     "eslint-scope": "5.1.1",
     "events": "^3.2.0",

--- a/test/configCases/typescript/alias-to-package-directory/index.js
+++ b/test/configCases/typescript/alias-to-package-directory/index.js
@@ -1,0 +1,5 @@
+"use strict";
+
+it("should resolve an aliased package directory whose name ends with .js", () => {
+	expect(require("vendor/pkg.js")).toBe("pkg.js");
+});

--- a/test/configCases/typescript/alias-to-package-directory/webpack.config.js
+++ b/test/configCases/typescript/alias-to-package-directory/webpack.config.js
@@ -1,0 +1,16 @@
+"use strict";
+
+const path = require("path");
+
+// The `.js` -> `.ts` extensionAlias that `experiments.typescript` installs must
+// not make a package directory named `pkg.js` unresolvable.
+
+/** @type {import("../../../../").Configuration} */
+module.exports = {
+	entry: "./index.js",
+	resolve: {
+		alias: {
+			vendor: path.resolve(__dirname, "node_modules")
+		}
+	}
+};

--- a/test/configCases/typescript/alias-to-package-directory/webpack.config.js
+++ b/test/configCases/typescript/alias-to-package-directory/webpack.config.js
@@ -3,7 +3,8 @@
 const path = require("path");
 
 // The `.js` -> `.ts` extensionAlias that `experiments.typescript` installs must
-// not make a package directory named `pkg.js` unresolvable.
+// not make a package directory named `pkg.js` unresolvable. Pinned here instead
+// of via `experiments.typescript` so the case also runs below Node.js 22.6.
 
 /** @type {import("../../../../").Configuration} */
 module.exports = {
@@ -11,6 +12,11 @@ module.exports = {
 	resolve: {
 		alias: {
 			vendor: path.resolve(__dirname, "node_modules")
+		},
+		extensionAlias: {
+			".js": [".js", ".ts"],
+			".cjs": [".cjs", ".cts"],
+			".mjs": [".mjs", ".mts"]
 		}
 	}
 };

